File: org.summer.view.widget/src/org/summer/view/widget/BaseValueSourceInternal.java
package org.summer.view.widget;
// Note that these enum values are arranged in the reverse order of 
// precendence for these sources. Local value has highest 
// precedence and Default value has the least. Note that we do not
// store default values in the _effectiveValues cache unless it is 
// being coerced/animated.
//[FriendAccessAllowed] // Built into Base, also used by Core & Framework.
/*internal*/ public enum BaseValueSourceInternal //: short
{ 
    Unknown                 ,//= 0,
    Default                 ,//= 1, 
    Inherited               ,//= 2, 
    ThemeStyle              ,//= 3,
    ThemeStyleTrigger       ,//= 4, 
    Style                   ,//= 5,
    TemplateTrigger         ,//= 6,
    StyleTrigger            ,//= 7,
    ImplicitReference       ,//= 8, 
    ParentTemplate          ,//= 9,
    ParentTemplateTrigger   ,//= 10, 
    Local                   ,//= 11, 
}

File: summer/WebContent/js/system/windows/BaseValueSourceInternal.js
define(["dojo/_base/declare"], function(declare){

    var BaseValueSourceInternal =  declare("BaseValueSourceInternal", null,{ 
    });

    BaseValueSourceInternal.Unknown                 = 0;
    BaseValueSourceInternal.Default                 = 1; 
    BaseValueSourceInternal.Inherited               = 2; 
    BaseValueSourceInternal.ThemeStyle              = 3;
    BaseValueSourceInternal.ThemeStyleTrigger       = 4; 
    BaseValueSourceInternal.Style                   = 5;
    BaseValueSourceInternal.TemplateTrigger         = 6;
    BaseValueSourceInternal.StyleTrigger            = 7;
    BaseValueSourceInternal.ImplicitReference       = 8; 
    BaseValueSourceInternal.ParentTemplate          = 9;
    BaseValueSourceInternal.ParentTemplateTrigger   = 10; 
    BaseValueSourceInternal.Local                   = 11;
    
    return BaseValueSourceInternal;
});